投稿機能（画像の追加）

aiorange19 opened the pull request:
・投稿に画像を表示する（複数枚）
・メイン画像とサブ画像を区別する

aiorange19 committed:
Categorymodel

aiorange19 committed:
item strongparameter

画像投稿の参考サイトでよくstrong parameterで作成されているため、まずすでに作成できていた画像以外の投稿部分でstrong parameterを作成。

aiorange19 committed:
img

ArakawaShintaro commented:
途中だったら
WIP 投稿機能（画像の追加）

ArakawaShintaro commented on Gemfile:
```diff
@@ -64,3 +64,7 @@ gem 'tzinfo-data', platforms: [:mingw, :mswin, :x64_mingw, :jruby]
 
 gem 'rails-i18n'
+
+gem 'pry-rails'
```
developmentにいれましょう

aiorange19 replied:
入れました

ArakawaShintaro commented on app/controllers/items_controller.rb:
```diff
@@ -13,15 +13,11 @@ def show
-        price: params[:item][:price],
-        user_id: @current_user.id
-        )
+    @item = set_current_user.items.build(item_params)
```
```ruby
@current_user.items.build(item_params)
```

aiorange19 replied:
@current_userに変更ずみです

ArakawaShintaro commented on app/models/image.rb:
```diff
@@ -0,0 +1,8 @@
+    belongs_to :item
+    mount_uploaders :image_name, ItemimageUploader
+   
+
```
無駄な空行

aiorange19 replied:
空行削除しました

ArakawaShintaro commented on app/views/items/index.html.erb:
```diff
@@ -3,7 +3,9 @@
-        <p><%= link_to(item.content, "/items/#{item.id}") %></p>
+        <p>商品説明：<%= link_to(item.content, "/items/#{item.id}") %></p>
+        <p>購入場所：<%= link_to(item.buy_place, "/items/#{item.id}") %></p>
+        <p>価格：<%= link_to(item.price, "/items/#{item.id}") %></p>
```
名前付きパスに変更しましょう

aiorange19 replied:
名前つきパスに変更ずみ（items/show.html.erbも)

ArakawaShintaro commented on app/views/items/show.html.erb:
```diff
@@ -5,7 +5,13 @@
+        購入場所：<%= @item.buy_place %>
+      </p>
+      <p>
+        価格：<%= @item.price %>
```
nullも考慮して見てください

aiorange19 replied:
・新規投稿時に購入場所を投稿しなかった場合「購入場所不明」と表示されるようにした。
・商品説明、価格に関してはバリデーションでnullにならないようにした

aiorange19 committed:
Item修正

指摘事項〜「名前付きパス」の所まで修正しました。

aiorange19 committed:
image update

・新規投稿作成時に、複数画像の投稿ができるようにした

aiorange19 committed:
Main_image

・メインイメージにtrue、他の画像にfalseが入るようにしました
・new.html.erbの1枚目の画像欄を「メイン画像」としました
・投稿一覧にメイン画像が表示されるようにしました

aiorange19 committed:
null buy_place

・新規投稿時に購入場所を投稿しなかった場合「購入場所不明」と表示されるようにした。
・商品説明、価格に関してはバリデーションでnullにならないようにした

ArakawaShintaro commented on app/controllers/items_controller.rb:
```diff
@@ -29,6 +29,7 @@ def create
 
     if @item.save
       flash[:notice] = "投稿を作成しました"
+        binding.pry
```
不要なので消しましょう

aiorange19 replied:
消しました

ArakawaShintaro commented on app/views/items/index.html.erb:
```diff
@@ -10,7 +10,13 @@
         <p>商品説明：<%= link_to(item.content, item_path(item.id)) %></p>
-        <p>購入場所：<%= link_to(item.buy_place, item_path(item.id)) %></p>
+        <p>
+          <% if item.buy_place == "" %>
```
`item.buy_place.blank?`
とかでいけそうです

aiorange19 replied:
`item.buy_place.blank?`に変更しました

ArakawaShintaro commented on app/controllers/items_controller.rb:
```diff
@@ -13,12 +13,20 @@ def show
+    
+    @item.images.each_with_index do |image, i|
+        if i == 0
+            image.main_image = "true"
```
`image.main_image = true`
trueが文字列である必要はなさそうです

aiorange19 replied:
こちらは、trueを文字列でなくすると、画像投稿時にtrue、falseが入力されなくなってしまいます。メイン画像が「１」、他画像が「０」が入ってしまいます。何か設定が間違えているのでしょうか？

aiorange19 committed:
correction

修正箇所コメント返しました

ArakawaShintaro commented on db/migrate/20190329133223_create_images.rb:
```diff
@@ -0,0 +1,11 @@
+  def change
+    create_table :images do |t|
+      t.string :image_name
+      t.string :main_image
```
あ、こちらstring型ではなく
boolean型にしましょう

aiorange19 replied:
型直しました！

aiorange19 committed:
correction2 boolean

・指摘箇所boolean型に変更しました
・true/falseの「””」を外しました

ArakawaShintaro commented on app/views/items/index.html.erb:
```diff
@@ -4,7 +4,7 @@
         <p><%= link_to(item.user.name, user_path(item.user.id)) %>
         <p><% item.images.each do |image| %>
-             <% if image.main_image == "true" %>
+             <% if image.main_image == true %>
```
boolean型なら
`image.main_image?` だけでいけそうです

aiorange19 replied:
変更し、特に問題なかったです！

ArakawaShintaro reviewed (approved):
下記の問題を修正し、動作確認して問題なければ、mergeしてOKです

aiorange19 committed:
correction3 image.main_image?

・image.main_image == trueを image.main_image?に変更

aiorange19 committed:
contribution item
